File: src/Page/LoginPage.java
package Page;

import SystemMessage.*;
import UserInfo.UserInfo;

public class LoginPage extends Page{

	public LoginPage(String content) {
		super(content);
		setType(PageType.LOGIN_PAGE);
		setName("Login Page");
	}


	@Override
	public SystemMessage runPage() {
		
		// print the login page content to console
		showContent();

		String userInfo[] = new String[2];	

		// take password
		System.out.println("Username:");
		userInfo[0] = takeInput();
		
		// take password
		System.out.println("Password:");
		userInfo[1] = takeInput();

		return new SystemMessage(FunctionType.LOGIN, null,userInfo);
	}	
}

File: python_development/Page/LoginPage.py
from python_development.Page.Page import Page
from python_development.Page.PageType import PageType
from python_development.System.FunctionType import FunctionType
from python_development.System.SystemMessage import SystemMessage


class LoginPage(Page):
    
    def __init__(self, content):
        super().__init__(content)
        self.setType("LOGIN_PAGE")
        self.setName("Login Page")

    
    def runPage(self):
        self.showContent()
        userInfo = [None, None]
        print("Username: ")
        userInfo[0] = self.takeInput()
        print("Password: ")
        userInfo[1] = self.takeInput()

        return SystemMessage(FunctionType.LOGIN, None,userInfo)